platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/AssertionConsumerService.java

```diff
@@ -274,6 +274,21 @@ public Response processSamlResponse(org.opensaml.saml.saml2.core.Response samlRe
                         relayUri.getQuery(),
                         relayUri.getFragment());
             }
+            //this avoids the user ever being redirected to the built in login page if they've already
+            //logged in via the IDP.
+            if ((relayUri.getPath()
+                    .equals("/login") || relayUri.getPath()
+                    .equals("/login/")) && (relayUri.getQuery() != null && relayUri.getQuery()
+                    .contains("prevurl"))) {
+                relayUri = new URI(relayUri.getScheme(),
+                        relayUri.getUserInfo(),
+                        relayUri.getHost(),
+                        relayUri.getPort(),
+                        relayUri.getQuery()
+                                .replace("prevurl=", ""),
```

Comment:
The `/login` path is actually the built-in login page for DDF, it is not the IdP page. This keeps you from getting redirected back to yet another login page after you've logged in via the IdP. I can show you what it looks like, if you want to see.